src/webui/api/torrentscontroller.cpp

```diff
@@ -250,6 +252,23 @@ namespace
             idList << BitTorrent::TorrentID::fromString(hash);
         return idList;
     }
+
+    nonstd::expected<QUrl, QString> validateWebseedUrl(const QString &urlStr)
+    {
+        // sanity check before parsing to prevent QUrl from checking for the existence of files on the filesystem
+        if (!supportedWebseedSchemes.contains(urlStr.section(u"://"_s, 0, 0)))
+            return nonstd::make_unexpected(TorrentsController::tr("URL scheme must be one of [%1]").arg(supportedWebseedSchemes.join(u",")));
+
+        const QUrl url(urlStr, QUrl::StrictMode);
+        if (!url.isValid())
+            return nonstd::make_unexpected(TorrentsController::tr("\"%1\" is not a valid url").arg(urlStr));
+
+        // sanity check after parsing in case QUrl interpereted a different scheme than intended
+        if (!supportedWebseedSchemes.contains(url.scheme()))
+            return nonstd::make_unexpected(TorrentsController::tr("URL scheme must be one of [%1]").arg(supportedWebseedSchemes.join(u",")));
```

Comment:
QUrl has support for reading local files, without (unfortunately) any explicit way to disable it. It appears that the only way to prevent QUrl from searching for files on the local filesystem is to ensure that a scheme is explicitly provided. I've taken the approach of supporting `http`, `https`, and `ftp`. This appears to align with [BEP 19](https://www.bittorrent.org/beps/bep_0019.html) and [BEP 17](https://www.bittorrent.org/beps/bep_0017.html), but wanted to call it out in case there are any other opinions. We can of course always modify this later if we run into issues.